packages/core/src/internal/fetchOptions.ts

```diff
@@ -40,7 +40,7 @@ export const makeFetchURL = (
     search.set('extensions', stringifyVariables(body.extensions));
 
   const finalUrl = url.toString();
-  if (finalUrl.length > 2047) {
+  if (finalUrl.length > 2047 && useGETMethod !== 'force') {
```

Comment:
Do we want to allow people to specify their own limit or are we just going to leave it to the escape-hatch?

Reply:
I don't really see personally why someone would have a different limit than the defacto IETF limit on the length of URLs, which also happens to be one that all browsers align on 🤔 
I think the main reason we're adding this is if someone knows their _entire_ stack can support variable length URLs, in which case, worst case, they're opting into HTTP stack errors, which are very observable. Most likely, I'd expect to see this in backend-only applications where likely no limit really applies.